web/src/main/java/org/cbioportal/web/GenericAssayController.java

```diff
@@ -48,6 +51,25 @@
     @Autowired
     private GenericAssayService genericAssayService;
 
+    @RequestMapping(value = "/generic_assay_meta/{genericAssayType}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)	
+    @ApiOperation("Get all meta data for one generic_assay_type")	
+    public ResponseEntity<List<GenericAssayMeta>> getGenericAssayMetaData(	
+        @ApiParam(required = true, value = "TREATMENT_RESPONSE")	
+        @PathVariable String genericAssayType,
+        @ApiParam("Level of detail of the response")	
+        @RequestParam(defaultValue = "SUMMARY") Projection projection,	
+        @ApiParam("Page size of the result list")	
+        @Max(Integer.MAX_VALUE)	
+        @Min(PagingConstants.MIN_PAGE_SIZE)	
+        @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,	
+        @ApiParam("Page number of the result list")	
+        @Min(PagingConstants.MIN_PAGE_NUMBER)	
+        @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber) throws GenericAssayNotFoundException {	
+
+        return new ResponseEntity<>(	
+            genericAssayService.getGenericAssayMetaByGenericAssayType(genericAssayType, projection.name(), pageSize, pageNumber), HttpStatus.OK);
+    }
+
     @PreAuthorize("hasPermission(#molecularProfileId, 'MolecularProfileId', 'read')")
```

Comment:
I noticed that the implementation of this routine is not quite right.  You'll need to introduce a interceptedGenericAssayMetaFilter.  Just a note - PreAuthorizing this method seems pointless in light of the fact that we are introducing /generic_assay_meta/{genericAssayType} without any preauthorization.  As stated above,  I guess the idea is that this information is just like gene information - which is freely available.  However, the fact that the data is requested via a molecular profile id is reason enough to keep preauthorization here.